client/src/App.js

```diff
@@ -57,28 +98,39 @@ class App extends Component {
       body.radius = 50;
     }
     return JSON.stringify(body);
-  }
+  };
 
-  searchHandler = (searchKeywords) => async() => {
-    var keywords = this.constructBody(searchKeywords);
+  searchHandler = async () => {
+    const keywords = this.constructBody(this.state.searchKeywords);
     await searchJobs(keywords);
     this.props.history.push('/results');
-  }
+  };
 
   render() {
+    const {
+      searchKeywords, selectedCategories, selectedIndustries, minSalary,
+    } = this.state;
+
     return (
       <div className="App">
-        <Navbar searchHandler={this.searchHandler}/>
-        {this.state.isFilterMenuVisible ?
-          <button id="filterButton" className="show clickable" onClick={this.toggleFilterVisibility}/> :
-          <button id="filterButton" className="hide clickable" onClick={this.toggleFilterVisibility}>Filters</button>}
-        <FilterMenu
-          visibility={this.state.isFilterMenuVisible}
-          changeSalary={this.onChangeMinSalary}
-          changeLocation={this.onChangeLocation}
+        <Navbar
+          searchKeywords={searchKeywords}
+          searchHandler={this.searchHandler}
+          updateSearchKeywords={this.updateSearchKeywords}
+        />
+        <Routes
+          selectedCategories={selectedCategories}
+          selectedIndustries={selectedIndustries}
+          minSalary={minSalary}
+          handleAddCategory={this.handleAddCategory}
+          handleAddIndustry={this.handleAddIndustry}
+          handleRemoveCategory={this.handleRemoveCategory}
+          handleRemoveIndustry={this.handleRemoveIndustry}
+          handleAddMinSalary={this.handleAddMinSalary}
+          handleRemoveMinSalary={this.handleRemoveMinSalary}
+          handleClearFilters={this.handleClearFilters}
```

Comment:
All of these are being passed down a lot and only used in components that are a few levels deep. Should I just add categories, industries, and salary to the store? Or should we maybe consider trying to add redux again?

Reply:
We ended up taking Redux out because it was more complicated than we needed. I'll say keep it like this for now, but we can consider Redux again if it would be useful for the auth part too